### PR TITLE
Nijil/wall 3711/wallets route update

### PR DESCRIPTION
**Pull request by nijil-deriv**
## Changes:
This PR is to make Trader's Hub with wallets and Trader's Hub without wallets to be in the same location, which is `app.deriv.com/`

- All routing and redirections to Deposit should be to `app.deriv.com/wallet/deposit`
- All routing and redirections to Withdrawal should be to `app.deriv.com/wallet/withdraw`
- All routing and redirections to Transfer should be to `app.deriv.com/wallet/account-transfer`
- All routing and redirections to Transactions should be to `app.deriv.com/wallet/transactions`
- All routing and redirections to Reset Balance should be to `app.deriv.com/wallets/reset-balance`

### CODE REVIEW

File: packages/shared/src/utils/routes/routes.ts [hunk 1]
```diff
@@ -78,13 +77,14 @@ export const routes = {
...
-    wallets_cashier: '/wallets/cashier',
-    wallets_deposit: '/wallets/cashier/deposit',
-    wallets_withdrawal: '/wallets/cashier/withdraw',
-    wallets_transfer: 'wallets/cashier/transfer',
-    wallets_transactions: '/wallets/cashier/transactions',
-    wallets_compare_accounts: '/wallets/compare-accounts',
+    wallets: '/wallet',
+    wallets_deposit: '/wallet/deposit',
+    wallets_withdrawal: '/wallet/withdrawal',
+    wallets_transfer: 'wallet/account-transfer',
```
Comment (lubega-deriv) [inline packages/shared/src/utils/routes/routes.ts]: https://github.com/binary-com/deriv-app/pull/15204
```suggestion
    wallets_transfer: '/wallet/account-transfer',
```

File: packages/appstore/src/components/routes/routes.tsx [hunk 2]
```diff
@@ -3,33 +3,37 @@ import { Loading } from '@deriv/components';
...
     const title_TH = localize("Trader's Hub");
     const title_TH_logged_out = localize('Deriv App');
 
-    React.useLayoutEffect(() => {
-        if (has_wallet) history.push(routes.wallets);
-    }, [history, has_wallet]);
+    const componentToRender = () => {
+        if (is_logged_in || is_logging_in) {
+            if (has_wallet) return Wallets;
+            return TradersHub;
```
Comment (heorhi-deriv) [inline packages/appstore/src/components/routes/routes.tsx]:
```suggestion
            return has_wallet ? Wallets : TradersHub;
```

File: packages/core/src/App/Components/Layout/Header/toggle-menu-drawer.jsx [hunk 3]
```diff
@@ -108,10 +107,10 @@ const ToggleMenuDrawer = observer(({ platform_config }) => {
 
             const location = window.location.pathname;
 
-            if (location === routes.traders_hub || is_trading_hub_category) {
-                primary_routes = [routes.account, routes.cashier];
-            } else if (has_wallet || location === routes.wallets) {
+            if (has_wallet && location === is_trading_hub_category) {
                 primary_routes = [routes.reports, routes.account];
+            } else if (location === is_trading_hub_category) {
+                primary_routes = [routes.account, routes.cashier];
```
Comment (heorhi-deriv) [inline packages/core/src/App/Components/Layout/Header/toggle-menu-drawer.jsx]:
```suggestion
            if (location === is_trading_hub_category) {
                primary_routes =  has_wallet ? [routes.reports, routes.account] : [routes.account, routes.cashier];
            }
```

File: packages/wallets/src/routes/Router.tsx [hunk 4]
```diff
@@ -70,40 +59,26 @@ declare module 'react-router-dom' {
...
+                path={'/compare-accounts'}
                 render={() => (
                     <React.Suspense fallback={<Loader />}>
                         <LazyCompareAccountsRoute />
                     </React.Suspense>
                 )}
             />
             <Route
-                path={`${walletsPrefix}/cashier`}
+                path={'/wallet'}
```
Comment (ghost) [inline packages/wallets/src/routes/Router.tsx:75]: is this a typo or intentional action? (/wallets vs /wallet)

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: packages/wallets/src/routes/Router.tsx (hunk 4 above)
Comment (nijil-deriv) [reply packages/wallets/src/routes/Router.tsx:75]: This is intentional. All fund management routes are now `/wallet/...` now.

**Review by heorhi-deriv — APPROVED** [review]

**Review by lubega-deriv — APPROVED** [review]

**Review by shahzaib-deriv — APPROVED** [review]

**Review by jim-deriv — APPROVED** [review]

**Review by sergei-deriv — APPROVED** [review]

**Pull request merged**